src/editors/textEditor.js

```diff
@@ -249,12 +269,34 @@ class TextEditor extends BaseEditor {
     this.textareaParentStyle.overflow = '';
     this.textareaParentStyle.position = '';
     this.textareaParentStyle.right = 'auto';
-    const zIndex = this.getEditedCellsZIndex();
-    this.textareaParentStyle.zIndex = zIndex !== 'auto' ? zIndex : '';
     this.textareaParentStyle.opacity = '1';
 
     this.textareaStyle.textIndent = '';
     this.textareaStyle.overflowY = 'hidden';
+
+    const childNodes = this.TEXTAREA_PARENT.childNodes;
+    let hasClassHandsontableEditor = false;
+
+    arrayFilter(childNodes, (childNode) => {
+      if (hasClass(childNode, 'handsontableEditor')) {
+        hasClassHandsontableEditor = true;
+      }
```

Comment:
Maybe use:

- [Array.find](https://developer.mozilla.org/pl/docs/Web/JavaScript/Referencje/Obiekty/Array/find) or 
- `rangeEach` with returning `false` when the class will be found (It would break the loop.). 

https://github.com/handsontable/handsontable/blob/98878a6bb6edbed065b7961bc97d4b799e908503/src/helpers/number.js#L35-L37

For me, `arrayFilter` isn't the best choice.